android-agent/src/main/java/io/opentelemetry/android/internal/services/applifecycle/ApplicationStateWatcher.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.services.applifecycle
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * This class is internal and not for public use. Its APIs are unstable and can change at any time.
+ */
+internal class ApplicationStateWatcher : DefaultLifecycleObserver {
```

Comment:
This class is meant to replace [this one](https://github.com/open-telemetry/opentelemetry-android/blob/5b80e00d41e83b6a1bf1d0c49667006dcc9fe600/android-agent/src/main/java/io/opentelemetry/android/ApplicationStateWatcher.java#L15).